### DISCUSSION TITLE
Like to run OpenEMS on a patch antenna under Windows.

### DISCUSSION THREAD

**Discussion by mistergitj** [thliebig/openEMS-Project#26] [General]
I am having a hard time getting traction on running Simple_Patch_Antenna under W10 x64. I have DLed most of what I think is important but can find no executable to run. Are there binaries available, or do I need to build one? Either way, please help me get started. Thank you, John.

**Comment by thliebig** [comment]
Hi, yeah I just realize that some of the install instructions and how to get started are not again available. I will have to look into that.

If you have downloaded the openEMS Windows zip file you really only need to unzip it to somewhere. Maybe to something like C:\openEMS

Inside there should certainly be an openEMS.exe but that is the solver directly and should not be just started like that.
What you need is [Octave](https://octave.org/) or Matlab because simulations are setup as Octave scripts. 
Octave needs to know where and how to find openEMS. Thus you need to add the path with:
`addpath('C:\opt\openEMS\matlab');`

Then you just can run e.g. the Tutorials provided in the "Matlab" folder.

If you want to user the python interface, have a look [here](https://docs.openems.de/python/install.html#windows).

I hope this helps?

**Comment by mistergitj** [comment]
Wow, thanks much! I will look into this especially Octave which I am not familiar with yet. You are correct, some of the links provided yield an HTTP 404. I searched the folder I extracted openEMS-master.zip into and there is no openEMS.exe, in fact no .exe files at all. What I am trying to do is analyze a patch antenna I used an online calculator to create. 4NEC2 gives wild numbers for input impedance and SWR. Thanks and enjoy, John.    On Wednesday, February 1, 2023 at 11:57:54 PM CST, Thorsten Liebig ***@***.***> wrote:  
 
 


Hi, yeah I just realize that some of the install instructions and how to get started are not again available. I will have to look into that.

If you have downloaded the openEMS Windows zip file you really only need to unzip it to somewhere. Maybe to something like C:\openEMS

Inside there should certainly be an openEMS.exe but that is the solver directly and should not be just started like that.
What you need is Octave or Matlab because simulations are setup as Octave scripts.
Octave needs to know where and how to find openEMS. Thus you need to add the path with:
addpath('C:\opt\openEMS\matlab');

Then you just can run e.g. the Tutorials provided in the "Matlab" folder.

If you want to user the python interface, have a look here.

I hope this helps?

**Comment by mistergitj** [comment]
Now I'm getting many errors running Simple_Patch_Antenna. Thanks, John.

   ----- Forwarded Message ----- From: John Christian ***@***.***>To: Thorsten Liebig ***@***.***>Sent: Thursday, February 2, 2023 at 06:08:14 PM CSTSubject: Fw: [thliebig/openEMS-Project] Like to run OpenEMS on a patch antenna under Windows. (Discussion #26)
  Found my problem. I should have DLed openEMS_x64_v0.0.35-86-g0777302_msvc. I have now and will continue to learn. Thanks, John.


I now have Octave installed but cannot figure out how to run the tutorials. Thanks, John.

   ----- Forwarded Message ----- From: John Christian ***@***.***>To: thliebig/openEMS-Project ***@***.***>Sent: Thursday, February 2, 2023 at 06:59:15 AM CSTSubject: Re: [thliebig/openEMS-Project] Like to run OpenEMS on a patch antenna under Windows. (Discussion #26)
  Wow, thanks much! I will look into this especially Octave which I am not familiar with yet. You are correct, some of the links provided yield an HTTP 404. I searched the folder I extracted openEMS-master.zip into and there is no openEMS.exe, in fact no .exe files at all. What I am trying to do is analyze a patch antenna I used an online calculator to create. 4NEC2 gives wild numbers for input impedance and SWR. Thanks and enjoy, John.    On Wednesday, February 1, 2023 at 11:57:54 PM CST, Thorsten Liebig ***@***.***> wrote:  
 
 


Hi, yeah I just realize that some of the install instructions and how to get started are not again available. I will have to look into that.

If you have downloaded the openEMS Windows zip file you really only need to unzip it to somewhere. Maybe to something like C:\openEMS

Inside there should certainly be an openEMS.exe but that is the solver directly and should not be just started like that.
What you need is Octave or Matlab because simulations are setup as Octave scripts.
Octave needs to know where and how to find openEMS. Thus you need to add the path with:
addpath('C:\opt\openEMS\matlab');

Then you just can run e.g. the Tutorials provided in the "Matlab" folder.

If you want to user the python interface, have a look here.

I hope this helps?

**Reply by thliebig** [reply]
Did you download the latest windows build? E.g. [here](https://github.com/thliebig/openEMS-Project/releases/latest)
There must be at least 2 *.exe and lots of *.dll inside... Otherwise you may have downloaded the source code only which will not run very well itself ;)

**Comment by mistergitj** [comment]
Thanks, I have now. At first I DLed Master now I have the one you pointed to. I am making progress, just not there yet. I have sent several e-mails but perhaps you are not getting them. The top link at the bottom of this link works but the other three yield HTTP 404 errors. I watched one of your videos yesterday and will look for more. Thanks, John.

    On Friday, February 3, 2023 at 12:29:40 AM CST, Thorsten Liebig ***@***.***> wrote:  
 
 


Did you download the latest windows build? E.g. here
There must be at least 2 *.exe and lots of *.dll inside... Otherwise you may have downloaded the source code only which will not run very well itself ;)

**Comment by mistergitj** [comment]
Got it running but it was one stumbling block after another. My main problem was getting the path defined in Octave for the folder where Physical_Constants was stored. I finnalt use the Octave ADDPATH command but had to change all \ into /. I was surprised by the time it took running all the time steps. Obviously more work si needed by me. Thanks, John.

   ----- Forwarded Message ----- From: John Christian ***@***.***>To: thliebig/openEMS-Project ***@***.***>Sent: Friday, February 3, 2023 at 06:01:51 AM CSTSubject: Re: [thliebig/openEMS-Project] Like to run OpenEMS on a patch antenna under Windows. (Discussion #26)
  Thanks, I have now. At first I DLed Master now I have the one you pointed to. I am making progress, just not there yet. I have sent several e-mails but perhaps you are not getting them. The top link at the bottom of this link works but the other three yield HTTP 404 errors. I watched one of your videos yesterday and will look for more. Thanks, John.

    On Friday, February 3, 2023 at 12:29:40 AM CST, Thorsten Liebig ***@***.***> wrote:  
 
 


Did you download the latest windows build? E.g. here
There must be at least 2 *.exe and lots of *.dll inside... Otherwise you may have downloaded the source code only which will not run very well itself ;)

**Comment by mistergitj** [comment]
Although running getting " Could not find the Qt platform plugin "windows" in "". StackOverflow has a really long answer to this, but I suspect there is a more appropriate method to correct it. Any help appreciated. Thanks and enjoy, John.

**Comment by mistergitj** [comment]
Please tell me how to increase the size of the Command window. Less than 20% of the window is used. My monitor is set to a resolution of 3820x2160. Thanks and enjoy, John.

**Reply by thliebig** [reply]
Hm that is a bit weird. Maybe you can try an [older version](https://github.com/thliebig/openEMS-Project/releases/tag/v0.0.35-74) that was not build using the MSVC.

Otherwise it should really just be setting the correct path with "addpath" and every tutorial should just run... nothing more to it...

**Reply by thliebig** [reply]
sorry that answer was meant for one question above ;)

**Comment by mistergitj** [comment]
Thanks much. Please see my posts for additional problems I am having. Thanks, John.

    On Saturday, February 4, 2023 at 07:26:46 AM CST, Thorsten Liebig ***@***.***> wrote:  
 
 


Hm that is a bit weird. Maybe you can try an older version that was not build using the MSVC.

Otherwise it should really just be setting the correct path with "addpath" and every tutorial should just run... nothing more to it...

**Comment by mistergitj** [comment]
I altered Simple_Patch_Antenna to represent my GPS L1 antenna. It runs and I like the output. Octave still complains about not find QT - don't know what I am missing. Thanks for all the help. I will continue working on my GPS antenna to refine if. Enjoy, John.

**Comment by myorangedragon** [comment]
I too am getting a problem with Qt. when I tried to upgrade to the latest version. This is being thrown up by CSXCAD which cannot find the windows platform plugin (presumably qwindows.dll?). I have tried giving it all the paths I can find with no success. If I run the CSXCAD executable from the command line I get:

 "qt.qpa.plugin: Could not find the Qt platform plugin "windows" in "c:\Qt\5.14.1\msvc2015_64\plugins\platforms\"
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem."

the "c:\Qt..."  path it is looking in is what I currently have QT_QPA_PLATFORM_PLUGIN_PATH environment variable set to.
I have tried building CSXCAD from sources but have run into all sorts of trouble ending in being told I need to have a microsoft compiler installed. I then tried to build from the python egg files but the error I get there is 

"C:\Users\Heather\AppData\Local\Programs\Python\Python311\python.exe: No module named easy_install"

I tried doing it with python310 as well with the same error. I think easy_install has been removed since PIP appeared?
Like mistergitj everything else runs (Octave, openEMS etc etc) and the final graphs and plots appear after the simulation  of one of the examples, but visualising the model in CSXCAD is, to put it mildly, rather important.

I have an older version running of openEMS (from 2019 using Qt4) and that works fine with CTXCAD.

Any help greatly appreciated. 

Thanks,
Heather

**Comment by myorangedragon** [comment]
FYI, the errors I got on building CSXCAD from source started with not having Cython installed (which I fixed with pip install cython). I then get this compiler error ...

 C:\openEMS\python>cd openEMS-Project/CSXCAD/python

C:\openEMS\python\openEMS-Project\CSXCAD\python>python setup.py build_ext -IC:openEMS\include -LC:openEMS
Compiling C:\openEMS\python\openEMS-Project\CSXCAD\python\CSXCAD\CSPrimitives.pyx because it changed.
Compiling C:\openEMS\python\openEMS-Project\CSXCAD\python\CSXCAD\CSProperties.pyx because it changed.
Compiling C:\openEMS\python\openEMS-Project\CSXCAD\python\CSXCAD\CSRectGrid.pyx because it changed.
Compiling C:\openEMS\python\openEMS-Project\CSXCAD\python\CSXCAD\CSTransform.pyx because it changed.
Compiling C:\openEMS\python\openEMS-Project\CSXCAD\python\CSXCAD\CSXCAD.pyx because it changed.
Compiling C:\openEMS\python\openEMS-Project\CSXCAD\python\CSXCAD\ParameterObjects.pyx because it changed.
Compiling C:\openEMS\python\openEMS-Project\CSXCAD\python\CSXCAD\Utilities.pyx because it changed.
[1/7] Cythonizing C:\openEMS\python\openEMS-Project\CSXCAD\python\CSXCAD\CSPrimitives.pyx
[2/7] Cythonizing C:\openEMS\python\openEMS-Project\CSXCAD\python\CSXCAD\CSProperties.pyx
[3/7] Cythonizing C:\openEMS\python\openEMS-Project\CSXCAD\python\CSXCAD\CSRectGrid.pyx
[4/7] Cythonizing C:\openEMS\python\openEMS-Project\CSXCAD\python\CSXCAD\CSTransform.pyx
[5/7] Cythonizing C:\openEMS\python\openEMS-Project\CSXCAD\python\CSXCAD\CSXCAD.pyx
[6/7] Cythonizing C:\openEMS\python\openEMS-Project\CSXCAD\python\CSXCAD\ParameterObjects.pyx
[7/7] Cythonizing C:\openEMS\python\openEMS-Project\CSXCAD\python\CSXCAD\Utilities.pyx
running build_ext
building 'CSXCAD.CSPrimitives' extension
error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/


HTH,
thanks,
Heather

**Reply by thliebig** [reply]
As the error says, you need the MS Compiler. I recommend to use a python for which I have per-compiled binaries. See [here](https://docs.openems.de/python/install.html#install-pre-build-modules)

**Reply by myorangedragon** [reply]
Hi Thorsten, 
Thanks for the super fast response and thanks for a great set of sim tools!
As I mentioned above, I did try to build using your pre-compiles egg files but couldn't find a way to get easy-install to run under windows. I got this error even after "pip install setuptools":

C:\Users\Heather\AppData\Local\Programs\Python\Python311\python.exe: No module named easy_install"

I couldn't find a way of getting the module now that pip has replaced such things.
Thanks,
Heather

**Comment by myorangedragon** [comment]
BTW,   openEMS_x64_v0.0.35-74-g0e54fbf.zip works fine for me.
Thanks,
Heather

**Reply by thliebig** [reply]
That indicates that a need to have a closer look at the MSVC builds on Windows... Somehow they ran fine for me but obviously not on every machine....

**Comment by mistergitj** [comment]
Heather, great to hear from a companion traveler. I am not nearly as far into the programmer type stuff as you are, but like you, I have openEMS/Octave working so I can see the plots for my patch antenna. Right now I am going back and reading the details to see what I can learn from the various manuals. Enjoy, John.

**Comment by myorangedragon** [comment]
Hi John, 
Yes, hello fellow traveller!!!! 
When you run you patch antenna sim, do you see the 3D CAD model of the antenna in CSXCAD? I think that is what you are missing out on when you get that Qt error. As I mentioned above, the version 35-74 works fine for me - giving the nice 3D image of what I am simulating before going on to do the actual sim and showing me the radiation plots etc. 
I find that the ability to see what I have created before I simulate it is very useful so it would be nice to get this latest version working with the 3D rendering. Still, I have upgraded from a 2019 version so a lot has happened since then that I can make sue of in the 35-74 version.
Regards,
Heather

**Comment by mistergitj** [comment]
Thanks, Heather. I agree with you that seeing my structure is important to me and suspect you are correct in that missing QT may be the root of the problem. I come from 4NEC2 (WW II electromagnetic solver) which always shows the structure first and it is also available in the 3D view of the emission pattern. The input impedance made no sense (almost totally imaginary). The numbers I get here are very reasonable and allow me feel better about building. There is a slight chance avoiding QT may be a blessing. I have encountered two really good applications using QT that take almost exactly 5 minutes to open. This started when I installed 22H2 in my W10. Thanks and enjoy, John.

**Comment by mistergitj** [comment]
What is needed, in my opinion, is a clear explanation of the environment variables needed for openEMS, Octave and QT. I do not even know the name of some of these variables, so it is very hard to set them. Also there are so many older versions of all the programs, searching for solutions often leads to stale answers. For us Windows user, we need to know information in Windows terminology not Linux. When we are instructed to do addpath, please say where. Thanks and enjoy, John.

**Comment by mistergitj** [comment]
Getting a little frustrating. I deleted all the files from 86, Dled and extracted 74. I ran my patch antenna based on simple_patch_antenna. In the command window I did:  addpath ("C:/openEMS/matlab/") then started the simulation. Immediately I got:
invoking AppCSXCAD, exit to continue script...
QCSXCAD - disabling editing
VTKPrimitives::AddCube: Warning, can't draw a Point Box... skipping
args = "patch_ant.xml"
 ----------------------------------------------------------------------
 | openEMS 64bit -- version v0.0.35-74-g0e54fbf
 | (C) 2010-2018 Thorsten Liebig <thorsten.liebig@gmx.de>  GPL license
 ----------------------------------------------------------------------
        Used external libraries:
                CSXCAD -- Version: v0.6.2-103-g3c4fb66
The opened window hung and after I killed it, the simulation continued and I got a Smith chart, followed by:
error: port(1): out of bound 0 (dimensions are 1x0)
error: called from
    plotRefl at line 130 column 9
    C:/OpenEMS/matlab/Tutorials/John'sGPS_Patch_Antenna.m at line 126 column 1
This is a new error and it ended the simulation. I used to get six plots now just one which is incomplete. Thanks for any help, John.

**Comment by mistergitj** [comment]
I had some real progress today. Still get the "...Point Box..." message, but it does not stop computation. I am getting the structure and it looks good. I got rid of the "...out of bound 0..." message by commenting out the plotRefl instruction and the following title. Now I get five plots (losing the Smith Chart). Turns out I know little about Smith Charts. The results look fairly good except for the feed point impedance. It is a lot more imaginary than I would like. Where can I see tabular data? I would love to receive some advice on reducing the imaginary part of the impedance. I clearly have something I can work with to help me understand. I would also like to get the Command window to be full height, width is OK. Thanks much and enjoy, John.

**Comment by mistergitj** [comment]
Heather, thank you for the information on older versions. Going to 'before MSVC' version 74 cured most of my problems. Thanks and enjoy, John.

**Comment by mistergitj** [comment]
This application is fun to use, so I started making what I thought were minor changes in the plotting area when I got:
 "error: source: error sourcing file 'C:/OpenEMS/matlab/Tutorials/John'sGPS_Patch_Antenna.m'" 
I chased it down to this line of code: 
plot( freq/1e6, (1+(abs(s11))/(1-(abs(s11)), 'g-', 'Linewidth', 2 );
The message is sometimes draconian because Octave will not let me change to edit tab to look at the code. Octave is not dead but I can do almost nothing and I have to close it. I have a good idea what is going and started to delete this post but reconsidered because I think this method of handling errors is not really acceptable. Thanks and enjoy, John.

**Comment by thliebig** [comment]
This discussion has gone a bit out of hand as everybody opened a new "thread" instead of answering to an existing one...

I have just tested the latest openEMS build with MSVC on a windows machine with Octave 7.3. Everything did run just fine.
What exactly seems to be the problem?

Note: If you want to reply to this, reply directly below it and not as a new answer/thread please.

**Comment by mistergitj** [comment]
Great to hear from you, thliebig! Not sure what you are referring to since really only Heather and I have been posting besides you.
Please tell me what version of openEMS and Windows version your are running. I am running W10 22H2 19045.2486 and now openEMS Build v0.0.35-74. I started with -86 but it had too many problems and you suggested to try -74 which runs much better. Some problems with -74:

Still some path problems, usually takes two or three tries at running a simulation, mainly path problems.

AppCSXCAD, hangs about once every three startups with 'not responding', killing it allows the simulation to continue to completion.

Command window will not fill panel but when I left click Window in the left panel then click 'Reset window default layout', the Command window goes to full width (still not in height) but goes back to narrow soon.

On some errors reported in the Command window, I can still do things in the command window but can not change windows (e.g. to Edit).

Thanks for your time, Enjoy, John.

**Reply by thliebig** [reply]
You again did not use the "reply" function but opened a new "comment/thread" ;)  I'm aware that this is a it weird here in github discussions... 

I tried it on some up to date Win10 and the latest Octave 7.3 and latest openEMS (-86 I think?). I just added the path to "C:\openEMS" using the menu "Edit" and "Set Path".

AppCSXCAD worked as expected. Everything else you are reporting sounds more like usability issues with Octave? I cannot really help with those...

br
Thorsten

**Reply by mistergitj** [reply]
Thanks, thliebig, I clicked on reply this time but can see nothing different. It sounds like you built a special version. What version of QT does it have? Can you please share that? Please tell me how do I get help on the Octave issues? Thanks and enjoy, John.

**Reply by thliebig** [reply]
Well it might be build with different Qt Versions for sure. But this should/must not interfere with Octave!
But I will have another check, e.g. if the Qt plugin and such are all in the right [place](https://github.com/thliebig/CSXCAD/blob/master/matlab/CSXGeomPlot.m#L46)

**Reply by thliebig** [reply]
Well this might indeed be it... The plugins are not in the new zip files... I will investigate... Strange that it just works for me...

**Reply by mistergitj** [reply]
Thanks much, thliebig! I will continue with -74 and would like to try the latest. Thanks and enjoy, John.

**Reply by thliebig** [reply]
I have uploaded a new build *-98. Please give it a try...

**Reply by mistergitj** [reply]
Thanks, thliebig. Will do. Enjoy, John.

**Reply by mistergitj** [reply]
Running now. Will let you know how how it goes, John.

**Reply by mistergitj** [reply]
Finished with no errors.

**Reply by mistergitj** [reply]
I was running this version, I thought:
<img width="1281" alt="35 74" src="https://user-images.githubusercontent.com/45215756/223548264-3b794f9f-2926-4933-92fe-4e400cbc7e79.png">
I DLed and extracted openEMS_x64_v0.0.35-99-gecf0c16_msvc and replaced the EMS folder in my openEMS folder. Then I ran my patch tutorial. Shortly after execution started, I see the 35.74 ,version as before not the 35.99. Obviously I did something wrong. Where can I look to see what version of open EMS I am running? Thanks and enjoy, John.

**Reply by mistergitj** [reply]
I found by problem but I would still like to know how to see the version of open EMS I am running. Thanks, john.

**Reply by mistergitj** [reply]
After correcting my paths still seeing version v0.0.35-74-g0e54fbf. Hope some one can help. Thanks and enjoy, John.

**Reply by thliebig** [reply]
open a cmd in windows and run the openEMS.exe itself. Maybe in a freshly unzipped download.
But if I have time later today I will check too if I maybe messed up the packaging... I had to repackage because Windows defender was complaining that the nf2ff.dll was or may contain a virus... which I could not figure out why...

**Reply by mistergitj** [reply]
Seems still seems to be my path problem. Here's what I get when I run openEMS in a command prompt:

----------------------------------------------------------------------
 | openEMS 64bit -- version v0.0.35-99-gecf0c16
 | (C) 2010-2023 Thorsten Liebig <thorsten.liebig@gmx.de>  GPL license
 ----------------------------------------------------------------------
        Used external libraries:
                CSXCAD -- Version: v0.6.2-123-gc29742b
                hdf5   -- Version: 1.12.0
                          compiled against: HDF5 library version: 1.12.0
                tinyxml -- compiled against: 2.6.2
                fparser
                boost  -- compiled against: 1_72
                vtk -- Version: 8.2.0
                       compiled against: 8.2.0
Thanks much. I will keep looking. Enjoy, John.

**Reply by mistergitj** [reply]
thliebig, I got 35-99 running. It was not installed properly in the first place. I deleted all the openEMS files and folders and uninstalled Octave, then installed Octave 7.3.0, DLed and extracted openEMS into C:/openEMS, did addpath "C:/openEMS/matlab/Tutorials" and addpath "C:/openEMS/matlab", then moved my modified Patch antenna .m file into the Tutorials folder and ran it. Looks really good! Thanks for all the help, Enjoy, John.

**Reply by thliebig** [reply]
Why do you put your own sim-files in the openEMS folder? Just put them in your user folder or somewhere else and keep the openEMS folder unchanged. This way you can always remove and replace it without much thought?

**Reply by mistergitj** [reply]
Good point, thliebig. I will do that. Thanks and enjoy, John.

**Reply by mistergitj** [reply]
thliebig, I put my patch antenna .m file in openEMS folder in Documents. When nf2ff phase begins I get these messages:

"CalcNF2FF: Reading nf2ff data only...
warning: load: 'C:\openEMS\matlab\Tutorials\tmp_Patch_Ant\nf2ff.h5' found by searching load path
warning: called from
    ReadHDF5Mesh>ReadHDF5Mesh_octave at line 57 column 5
    ReadHDF5Mesh at line 19 column 14
    ReadNF2FF at line 15 column 10
    CalcNF2FF at line 111 column 11
    JohnsGPS_Patch_Antenna at line 222 column 7

error: h5readatt_octave: opening the given File failed
error: called from
    ReadHDF5Attribute at line 22 column 10
    ReadNF2FF at line 22 column 12
    CalcNF2FF at line 111 column 11
    JohnsGPS_Patch_Antenna at line 222 column 7"

My paths:
![SetPath](https://user-images.githubusercontent.com/45215756/223967181-44fd05a2-603a-4fac-bec1-8e4ab209abec.png)
Thanks, John.

**Reply by mistergitj** [reply]
thliebig, I went back and ran the same .m file from Tutorials folder and it ran fine. Some comments on 35-99:

It is a little faster than 35-74. It seems I remember -74 taking about 3 minutes and -99 takes 30 seconds to about 1 minute. In one case it says 4 threads and the other 2 threads optional. I have a utility for my processor (AMD 3970X) to see what each core is doing. In the first case one core is running about 4 GHz and other 3 are around 1 GHz each. In the other case each core is at about 2 GHz. If you like I can post any requested screenshots. Here's one:
![2Threads](https://user-images.githubusercontent.com/45215756/224062511-a6e4e2f3-7d5c-4b06-a896-2921774c953e.png)
The problem with the Command window on filling less than half vertically seems to be fixed.
I like it, thanks so much.
Enjoy, John.

**Reply by mistergitj** [reply]
thliebig, I am confused by the simulation Multi-threaded results (35-99). The total time changes from run to run. With four threads total time is about 30 seconds and other times it is almost three minutes. Would like to hear some comments.
Another question; I used the zeros command as follows: TP = Zeros (1,2,501); and get 'Zeros' undefined. What am I doing wrong and where should I find the answer. I found Zeros in the Documentation tab of Octave and think I am using it correctly. The command works in the Command window. Thanks for your help, Enjoy, John.

**Reply by thliebig** [reply]
Well speed depends on what your PC is doing in the background. And maybe this auto-detect of threads is not working too well yet too?
I think the command is "zeros" not with any capital letters?

**Reply by mistergitj** [reply]
Thanks, thliebig. You are correct on zeros and not Zeros. I usually am only running a browser playing music. I closed all that I could and it made little difference: 2 threads and 1 min 40 seconds. Enjoy, John.

**Reply by mistergitj** [reply]
I hate to keep asking questions here, but do not know a better place. Please let me know where I can get help. My latest problems are errors that stop the simulations:

![SimErrors](https://user-images.githubusercontent.com/45215756/225023718-bb6e47b7-54d1-4308-86bd-76bb81445fdc.png)

I am running Simple Patch Antenna Tutorial on openEMS  64bit -- version v0.0.35-99-gecf0c16 and Octave-8.1.0. I started getting these warnings and errors on my modified version and could not resolve them so I ran the unaltered tutorial and got the same messages. It could be a path problem. Thanks for any help. Enjoy, John.

**Reply by thliebig** [reply]
Well no, the discussion explicitly is meant to ask such questions.

But I'm not sure what is going on with your installation? Maybe you try to run "setup" to recompile the h5read_att module.
Otherwise you maybe move/rename the folder and try with a completely fresh install?

**Reply by mistergitj** [reply]
Thanks, thliebig. I will continue to post here. I will try some installation modifications. On another subject, I ran some test on multi-threading with these results:
Threads   Time Seconds
 1                  32
 2                113
 3                  32
 4                117
 5                  85
 6                  28
 8                117
20               146
50               179     
Thanks and enjoy, John.

**Reply by mistergitj** [reply]
I looked at the hash codes for h5read_att and all looked good but then I ran the setup and now I get a different error on my altered patch:

"error: h5readatt_octave: opening the given File failed
error: called from
    ReadHDF5Attribute at line 22 column 10
    ReadNF2FF at line 22 column 12
    CalcNF2FF at line 111 column 11
    JohnsGPS_Patch_Antenna at line 209 column 7" 

Code at line 209 is: "nf2ff = CalcNF2FF(nf2ff, Sim_Path, freq , 0.0 , 0.0 );"

I also ran an unaltered Horn and Simple_Patch_Antenna. Simple_Patch_Antenna gave the same error as my altered one. The horn tutorial ran without error. 
This makes me wondered if I am supposed to run setup after updating and are there more 'startup' scripts or .exe files I need to run. Thanks much for your help. Enjoy, John.

**Reply by mistergitj** [reply]
I went back to Octave 7.3.0 but still at openEMS 35-99. Starting to work and now I think I am seeing errors I have created. Will continue to debug. Thanks and enjoy, John.

**Reply by mistergitj** [reply]
I may have a hint to the problem. The module "C:\OpenEMS\matlab\h5readatt_octave.oct" is being left with an outstanding Handle open:
![h5readatt_octave](https://user-images.githubusercontent.com/45215756/225472002-3f990a95-c79b-41f9-a0f0-079ca2fcddd8.png)
This is after the error error that stops the simulation. I need to close Octave to clear the Handle. please let me know what you think. Thanks and enjoy, John.
